src/main/java/programminglife/parser/GraphParser.java

```diff
@@ -50,15 +45,17 @@ public GraphParser(File graphFile) throws IOException {
     public void run() {
         try {
             System.out.printf("[%s] Parsing GenomeGraph on separate Thread%n", Thread.currentThread().getName());
+            long startTime = System.nanoTime();
             parse(this.verbose);
-            DataManager.commit();
 
-            int secondsElapsed = (int) ((System.nanoTime() - this.startTime) / 1000000000.d);
+            int secondsElapsed = (int) ((System.nanoTime() - startTime) / 1000000000.d);
             System.out.printf("[%s] Parsing took %d seconds%n", Thread.currentThread().getName(), secondsElapsed);
             this.setChanged();
             this.notifyObservers(this.graph);
         } catch (Exception e) {
-            DataManager.rollback();
+            try {
+                this.getGraph().rollback();
+            } catch (IOException eio) { }
```

Comment:
Shouldn't something happen when this goes wrong? For example, alert the user and restart the application?